demos/howtos/1d_deconvolution_mrf_priors.py

```diff
@@ -0,0 +1,542 @@
+#!/usr/bin/env python3
+"""
+1D Deconvolution with MRF Priors: GMRF, CMRF, and LMRF
+
+This demo recreates the 1D deconvolution examples from the CUQIpy paper,
+showcasing three different Markov Random Field priors:
+
+1. GMRF (Gaussian Markov Random Field) - promotes smoothness
+2. CMRF (Cauchy Markov Random Field) - preserves edges while smoothing
+3. LMRF (Laplace Markov Random Field) - promotes sparsity in differences
+
+The example demonstrates:
+- 1D deconvolution problem setup
+- Different prior behaviors and their effects
+- Backend switching between NumPy and PyTorch
+- Gradient computation capabilities with PyTorch
+- Bayesian inference and uncertainty quantification
+"""
+
+import os
+import sys
+import numpy as np
+import matplotlib.pyplot as plt
+from contextlib import contextmanager
+
+@contextmanager
+def backend_context(backend_name):
```

Comment:
Please make this whole howto much simpler and do not use this many functions. Instead make it script-like. Showcase and plit using the built-in plotting in CUQIpy. Switch between Numpy and PyTorch to show it gives the same results.